### ISSUE TITLE
attributes Instrument,total size of experiment are not present in the metadata fetched through flags

### ISSUE THREAD

**Issue by deepthi888** [saketkc/pysradb#18]
Hi @saketkc,
could you please provide me information on how to fetch the attribute "Instrument" under "Library"  
 and also the total size of experiment by giving bio project id as input.
I tried with this "pysradb metadata SRP063732 --desc --expand --detailed" fetched layout,strategy,name,source,selection  under Library but not "Instrument" and also,it fetched only bases and spots but not the total size of experiment

Thanks in advance.

**Comment by saketkc** [comment]
Hi @deepthi888,

Sorry for the delay in my response. 

Thank you for pointing this out. It was indeed missing from the metadata response.

The [recent](https://github.com/saketkc/pysradb/releases/tag/v0.9.7) release prints instrument and total size by default in the SRAweb mode. I am slowly planning to retire the SRAdb mode, which means we will no longer require the SQLite file as it seems to be no longer updated and is missing a lot of SRA projects.

Here is the output for your query:
```
> pysradb metadata SRP063732 --desc --detailed --expand
experiment_accession experiment_title                                                       experiment_desc                                                        organism_taxid  organism_name   library_strategy library_source library_selection instrument           total_spots total_size   sample_accession sample_title study_accession run_accession run_total_spots run_total_bases
 SRX1249861           Bisulfite-seq of rhesus macaque 26168_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 26168_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  89051187    11830170462  SRS1071080                    SRP063732       SRR2420581    89051187        17988339774   
 SRX1249860           Bisulfite-seq of rhesus macaque 26148_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 26148_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  87335402    11763248173  SRS1071081                    SRP063732       SRR2420580    87335402        17641751204   
 SRX1249859           Bisulfite-seq of rhesus macaque 25822_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 25822_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83544130    9481342835   SRS1071082                    SRP063732       SRR2420579    83544130        13784781450   
 SRX1249858           Bisulfite-seq of rhesus macaque 25154_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 25154_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  115476876   13003165573  SRS1071083                    SRP063732       SRR2420578    115476876       19053684540   
 SRX1249857           Bisulfite-seq of rhesus macaque 23784_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23784_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  101539017   13697691917  SRS1071084                    SRP063732       SRR2420577    101539017       20510881434   
 SRX1249856           Bisulfite-seq of rhesus macaque 23779_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23779_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  98394582    13326784419  SRS1071085                    SRP063732       SRR2420576    98394582        19875705564   
 SRX1249855           Bisulfite-seq of rhesus macaque 23773_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23773_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84442355    11218164664  SRS1071086                    SRP063732       SRR2420575    84442355        17057355710   
 SRX1249854           Bisulfite-seq of rhesus macaque 23764_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23764_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  87197006    11291883405  SRS1071087                    SRP063732       SRR2420574    87197006        17613795212   
 SRX1249853           Bisulfite-seq of rhesus macaque 23762_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23762_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  89455608    11917710767  SRS1071088                    SRP063732       SRR2420573    89455608        18070032816   
 SRX1249852           Bisulfite-seq of rhesus macaque 25811_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25811_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  88216999    10285962574  SRS1071089                    SRP063732       SRR2420572    88216999        14555804835   
 SRX1249851           Bisulfite-seq of rhesus macaque 25790_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25790_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  94731241    10748979142  SRS1071090                    SRP063732       SRR2420571    94731241        15630654765   
 SRX1249850           Bisulfite-seq of rhesus macaque 25742_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25742_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  113512952   17943763654  SRS1071091                    SRP063732       SRR2420570    113512952       22929616304   
 SRX1249849           Bisulfite-seq of rhesus macaque 25240_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25240_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  81002663    9232986327   SRS1071092                    SRP063732       SRR2420569    81002663        13365439395   
 SRX1249848           Bisulfite-seq of rhesus macaque 25184_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25184_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  110228135   12594727479  SRS1071093                    SRP063732       SRR2420568    110228135       18187642275   
 SRX1249847           Bisulfite-seq of rhesus macaque 25157_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25157_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  97249804    11026452312  SRS1071094                    SRP063732       SRR2420567    97249804        16046217660   
 SRX1249846           Bisulfite-seq of rhesus macaque 21119_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 21119_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  68899204    9241244635   SRS1071095                    SRP063732       SRR2420566    68899204        13917639208   
 SRX1249845           Bisulfite-seq of rhesus macaque 26104_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 26104_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84859783    9604170375   SRS1071096                    SRP063732       SRR2420565    84859783        14001864195   
 SRX1249844           Bisulfite-seq of rhesus macaque 26082_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 26082_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  73346886    8248399265   SRS1071097                    SRP063732       SRR2420564    73346886        12102236190   
 SRX1249843           Bisulfite-seq of rhesus macaque 25207_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 25207_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  77306973    12235827725  SRS1071098                    SRP063732       SRR2420563    77306973        15616008546   
 SRX1249842           Bisulfite-seq of rhesus macaque 25407_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 25407_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83985968    13355795218  SRS1071099                    SRP063732       SRR2420562    83985968        16965165536   
 SRX1249841           Bisulfite-seq of rhesus macaque 24818_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24818_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  72782928    8484613123   SRS1071100                    SRP063732       SRR2420561    72782928        12009183120   
 SRX1249840           Bisulfite-seq of rhesus macaque 24698_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24698_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  79414016    12627204820  SRS1071101                    SRP063732       SRR2420560    79414016        16041631232   
 SRX1249839           Bisulfite-seq of rhesus macaque 24340_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24340_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84772576    13436053136  SRS1071102                    SRP063732       SRR2420559    84772576        17124060352   
 SRX1249838           Bisulfite-seq of rhesus macaque 22770_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 22770_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83310630    13224887149  SRS1071103                    SRP063732       SRR2420558    83310630        16828747260   

```

**Comment by deepthi888** [comment]
Thanks for the response.
could you please let me know how to use this SRAweb mode in google colaboratory

thanks in advance.

**Comment by saketkc** [comment]
Here's one demonstrating relevant commands: https://colab.research.google.com/drive/1C60V-jkcNZiaCra_V5iEyFs318jgVoUR

**Comment by deepthi888** [comment]
thanks for your response@saketkc

**Comment by deepthi888** [comment]
@saketkc , could you please let me know about how to fetch attributes of a sample such as isolate, development stage,sex,tissue and treatment

**Comment by saketkc** [comment]
Currently missing from the SRAweb version, but I just got it working locally so it should be in the master branch sometime this week. Thank you for your patience!

**Comment by saketkc** [comment]
Hi @deepthi888, the latest commit on master bracnh will allow you to fetch these attributes:

```
> pysradb metadata SRP063732 --detailed
experiment_accession experiment_title                                                       experiment_desc                                                        organism_taxid  organism_name   library_strategy library_source library_selection instrument           total_spots total_size   sample_accession sample_title study_accession run_accession run_total_spots run_total_bases isolate                 dev_stage sex   tissue             treatment     BioSampleModel           
 SRX1249861           Bisulfite-seq of rhesus macaque 26168_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 26168_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  89051187    11830170462  SRS1071080                    SRP063732       SRR2420581    89051187        17988339774     26168_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249860           Bisulfite-seq of rhesus macaque 26148_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 26148_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  87335402    11763248173  SRS1071081                    SRP063732       SRR2420580    87335402        17641751204     26148_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249859           Bisulfite-seq of rhesus macaque 25822_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 25822_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83544130    9481342835   SRS1071082                    SRP063732       SRR2420579    83544130        13784781450     25822_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249858           Bisulfite-seq of rhesus macaque 25154_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 25154_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  115476876   13003165573  SRS1071083                    SRP063732       SRR2420578    115476876       19053684540     25154_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249857           Bisulfite-seq of rhesus macaque 23784_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23784_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  101539017   13697691917  SRS1071084                    SRP063732       SRR2420577    101539017       20510881434     23784_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249856           Bisulfite-seq of rhesus macaque 23779_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23779_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  98394582    13326784419  SRS1071085                    SRP063732       SRR2420576    98394582        19875705564     23779_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249855           Bisulfite-seq of rhesus macaque 23773_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23773_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84442355    11218164664  SRS1071086                    SRP063732       SRR2420575    84442355        17057355710     23773_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249854           Bisulfite-seq of rhesus macaque 23764_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23764_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  87197006    11291883405  SRS1071087                    SRP063732       SRR2420574    87197006        17613795212     23764_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249853           Bisulfite-seq of rhesus macaque 23762_HeavyDrinker: Nucleus accumbens  Bisulfite-seq of rhesus macaque 23762_HeavyDrinker: Nucleus accumbens  9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  89455608    11917710767  SRS1071088                    SRP063732       SRR2420573    89455608        18070032816     23762_NucleusAccumbens  Adult     male  Nucleus accumbens  HeavyEthanol  Model organism or animal
 SRX1249852           Bisulfite-seq of rhesus macaque 25811_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25811_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  88216999    10285962574  SRS1071089                    SRP063732       SRR2420572    88216999        14555804835     25811_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249851           Bisulfite-seq of rhesus macaque 25790_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25790_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  94731241    10748979142  SRS1071090                    SRP063732       SRR2420571    94731241        15630654765     25790_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249850           Bisulfite-seq of rhesus macaque 25742_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25742_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  113512952   17943763654  SRS1071091                    SRP063732       SRR2420570    113512952       22929616304     25742_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249849           Bisulfite-seq of rhesus macaque 25240_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25240_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  81002663    9232986327   SRS1071092                    SRP063732       SRR2420569    81002663        13365439395     25240_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249848           Bisulfite-seq of rhesus macaque 25184_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25184_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  110228135   12594727479  SRS1071093                    SRP063732       SRR2420568    110228135       18187642275     25184_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249847           Bisulfite-seq of rhesus macaque 25157_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 25157_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  97249804    11026452312  SRS1071094                    SRP063732       SRR2420567    97249804        16046217660     25157_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249846           Bisulfite-seq of rhesus macaque 21119_LowDrinker: Nucleus accumbens    Bisulfite-seq of rhesus macaque 21119_LowDrinker: Nucleus accumbens    9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  68899204    9241244635   SRS1071095                    SRP063732       SRR2420566    68899204        13917639208     21119_NucleusAccumbens  Adult     male  Nucleus accumbens  LightEthanol  Model organism or animal
 SRX1249845           Bisulfite-seq of rhesus macaque 26104_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 26104_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84859783    9604170375   SRS1071096                    SRP063732       SRR2420565    84859783        14001864195     26104_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249844           Bisulfite-seq of rhesus macaque 26082_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 26082_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  73346886    8248399265   SRS1071097                    SRP063732       SRR2420564    73346886        12102236190     26082_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249843           Bisulfite-seq of rhesus macaque 25207_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 25207_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  77306973    12235827725  SRS1071098                    SRP063732       SRR2420563    77306973        15616008546     25207_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249842           Bisulfite-seq of rhesus macaque 25407_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 25407_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83985968    13355795218  SRS1071099                    SRP063732       SRR2420562    83985968        16965165536     25407_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249841           Bisulfite-seq of rhesus macaque 24818_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24818_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  72782928    8484613123   SRS1071100                    SRP063732       SRR2420561    72782928        12009183120     24818_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249840           Bisulfite-seq of rhesus macaque 24698_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24698_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  79414016    12627204820  SRS1071101                    SRP063732       SRR2420560    79414016        16041631232     24698_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249839           Bisulfite-seq of rhesus macaque 24340_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 24340_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  84772576    13436053136  SRS1071102                    SRP063732       SRR2420559    84772576        17124060352     24340_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
 SRX1249838           Bisulfite-seq of rhesus macaque 22770_Control: Nucleus accumbens       Bisulfite-seq of rhesus macaque 22770_Control: Nucleus accumbens       9544            Macaca mulatta  Bisulfite-Seq    GENOMIC        other             Illumina HiSeq 2000  83310630    13224887149  SRS1071103                    SRP063732       SRR2420558    83310630        16828747260     22770_NucleusAccumbens  Adult     male  Nucleus accumbens  Control       Model organism or animal
```

You can install the latest version using:
`pip install git+https://github.com/saketkc/pysradb.git`

Let me know if you face any issues.

**Comment by saketkc** [comment]
Hi @deepthi888,

This is now available in [v0.9.9](https://github.com/saketkc/pysradb/releases/tag/v0.9.9).
Please let me know if you run into issues.

Here's a colab version demonstrating `--detailed` output: https://colab.research.google.com/drive/1C60V-jkcNZiaCra_V5iEyFs318jgVoUR
